src/main/java/uk/ac/ebi/ait/filecontentvalidatorservice/service/FileContentValidationHandler.java

```diff
@@ -91,6 +121,70 @@ public ValidationResponse handleFileContentValidation() {
         return validationResponse;
     }
 
+    public void sendValidationMessagesToAggregator(ValidationResponse validationResponse) {
+        commandLineParameters.getFilesData().forEach(fileParameters -> {
+            List<SingleValidationResult> validationResultByFileUUID =
+                    createValidationResultByFileUUID(validationResponse, fileParameters.getFileUUID());
+            SingleValidationResultsEnvelope singleValidationResultsEnvelope =
+                    generateSingleValidationResultsEnvelope(validationResultByFileUUID,
+                            Integer.parseInt(fileParameters.getValidationResultVersion()),
+                            fileParameters.getValidationResultUUID());
+            sendValidationMessageToAggregator(singleValidationResultsEnvelope);
+        });
+    }
+
+    public List<SingleValidationResult> createValidationResultByFileUUID(ValidationResponse validationResponse, String fileUUID) {
+        List<SingleValidationResult> validationResults = new ArrayList<>();
+        if (validationResponse.getStatus().equals(ValidationResponse.status.VALIDATION_SUCCESS)) {
+            validationResults.add(SingleValidationResultBuilder.buildSingleValidationResultWithPassStatus(fileUUID));
+        } else {
+            validationResults.addAll(parseResultFiles(fileUUID));
+        }
+
+        return validationResults;
+    }
+
+    private List<SingleValidationResult> parseResultFiles(String fileUUID) {
+        List<SingleValidationResult> validationResults = new ArrayList<>();
+
+        final SubmissionFile submissionFile = submissionFileByFileUUID.get(fileUUID);
+        if (submissionFile == null) {
+            throw new FileContentValidationException(String.format(SUBMISSION_FILE_COULD_NOT_BE_FOUND, fileUUID));
+        }
+
+        validationResults.addAll(parseResultFile(fileUUID, submissionFile.getReportFile()));
+
+        validationResults.addAll(parseCommonResultFile(fileUUID));
+
+        return validationResults;
+    }
+
+    private List<SingleValidationResult> parseCommonResultFile(String fileUUID) {
+        List<SingleValidationResult> validationResults = new ArrayList<>();
+        validationResults.addAll(parseResultFile(fileUUID, getSubmissionReportFile()));
+
+        return validationResults;
+    }
+
+    private List<SingleValidationResult> parseResultFile(String fileUUID, File reportFile) {
+        List<SingleValidationResult> validationResults = new ArrayList<>();
+        try (Scanner scanner = new Scanner(reportFile)) {
+            while (scanner.hasNext()) {
+                String message = scanner.nextLine();
```

Comment:
Is there any reason for not using the BufferedReader? It works really well with large files too.

Reply:
Thanks for this, I pushed a fix